### ISSUE TITLE
[Feature] Add `op.#ApplyRemaining` to apply remaining resources

### ISSUE THREAD

**Issue by hongchaodeng** [oam-dev/kubevela#1980]
## Background

In application workflow, user can specify what resource to apply via `op.#Apply`. While this is very flexible, there are some cases it could solve elegantly:

- Conventionally, all resources are applied automatically and users might not be aware of any specific resources. Users would prefer not to write down all names of the resources to apply
- In workflow steps, while some steps might deal with specific components and traits (e.g. rolling upgrade, traffic splitting), users might want to apply all the remaining resources not applied in previous steps without knowing them all.
- This is an additional requirement: In multi-cluster setup, users might want to apply all remaining resources by namespace.

## Proposal

We propose to add a new action to built-in CUE library:

```yaml
#ApplyRemaining: {

  namespace: string

  // exceptions specify the resources not to apply.
  exceptions: [

    [componentName: string]: {
      // skipApplyWorkload indicates whether to skip apply the workload resource
      skipApplyWorkload: *true | bool
      
      // skipAllTraits indicates to skip apply all resources of the traits.
      // If this is true, skipApplyTraits will be ignored
      skipAllTraits: *true| bool

      // skipApplyTraits specifies the names of the traits to skip apply
      skipApplyTraits: [...string]
    }
  ]
}
```

Users specify the resources they want to skip apply so that all the remaining resources that they are not aware of will be applied automatically. For example, when dealing with data passing case A->B, users will have to apply A and B step by step to ensure ordering and waiting. This is special components that users should be aware of. But after that, all the remaining resources could be applied without understanding more details.

**Comment by wonderflow** [comment]
The user experience looks pretty good! Let's see how to implement it.

**Comment by leejanee** [comment]
Realized by cue configuration
```
#ApplyRemaining: #Steps & {
  namespace?: string

  // exceptions specify the resources not to apply.
  exceptions?: {
    [componentName: string]: {
      // skipApplyWorkload indicates whether to skip apply the workload resource
      skipApplyWorkload: *true | bool
      
      // skipAllTraits indicates to skip apply all resources of the traits.
      // If this is true, skipApplyTraits will be ignored
      skipAllTraits: *true| bool

      // skipApplyTraits specifies the names of the traits to skip apply
      skipApplyTraits: [...string]
    }
  }
  
  list: #ListObjects
  applied: #GetVar & {
      path: "applied__"
  } 
  for i , o in list.objects {
     if applied.value["\(o.metadata.namespace)_\(o.apiVersion)_\(o.kind)_\(o.metadata.name)"]==_|_{
        "s\(i)": #Apply & {object: o}
     }
  }
}
```

**Comment by wonderflow** [comment]
fixed by #1993
